lib/pdn/pdn.c

```diff
@@ -598,6 +599,37 @@ int pdn_id_get(uint8_t cid)
 	return strtoul(p + 1, NULL, 10);
 }
 
+int pdn_dynamic_params_get(uint8_t cid, struct in_addr *dns4_pri,
+			   struct in_addr *dns4_sec, unsigned int *ipv4_mtu)
+{
+	int ret;
+	char resp[256];
+	const char *fmt;
+	int dns4_pri_offset;
+	int dns4_sec_offset;
+
+	ret = nrf_modem_at_cmd(resp, sizeof(resp), "AT+CGCONTRDP=%u", cid);
+	if (ret) {
+		LOG_ERR("Failed to read dynamic params for CID %u, err %d", cid, ret);
+		return ret;
+	}
+	   /* "+CGCONTRDP: 0,,"example.com","","","198.276.154.230","198.276.154.203",,,,,1464" */
+	fmt = "+CGCONTRDP: %*u,,\"%*[^\"]\",\"\",\"\",\"%n%*15[0-9.]\",\"%n%*15[0-9.]\",,,,,%u";
+	ret = sscanf(resp, fmt, &dns4_pri_offset, &dns4_sec_offset, ipv4_mtu);
+	if (ret != 1) {
+		return -EBADMSG;
+	}
```

Comment:
Why don't you read the addresses into a string buffer?
It looks a lot cleaner.

```suggestion
	char pri[IPV6_ADDR_STR_LEN];
	char sec[IPV6_ADDR_STR_LEN];
	char *fmt = "+CGCONTRDP: %*u,,\"%*[^\"]\",\"\",\"\",\"%[^\"]\",\"%[^\"]\",,,,,%u";

	ret = sscanf(resp, fmt, &pri, &sec, &ipv4_mtu);
	if (ret != 3) {
		printf("EBADMSG\n");
		return -1;
	}
	inet_pton(AF_INET, pri, dns4_pri); // ...
```